src/TTextEdit.cpp

```diff
@@ -766,11 +766,16 @@ void TTextEdit::highlight()
         if (y == y1) {
             x = mPA.x();
         }
+
+        if (y >= static_cast<int>(mpBuffer->buffer.size())) {
+            break;
+        }
+        mpBuffer->dirty[y] = true;
+
         for (;; x++) {
             if ((y == mPB.y()) && (x > mPB.x())) {
                 break;
             }
-            mpBuffer->dirty[y] = true;
```

Comment:
... instead we need to revise this idempotent (now that _is_ a good word) line but move it down into the next bit of code because the _line_ only needs to be marked _dirty_ if _any_ part of it has the corresponding text attribute (`mpBuffer->buffer[y][x]`) _modified_ i.e. that `mpBuffer->buffer[y][x].flags & TCHAR_INVERSE` is false before old/new line 775/780 is reached...!  The `unhighlight` method is also adrift in this area. 🤦‍♂️